webhook.rst

```diff
@@ -0,0 +1,279 @@
+    namespace App\RemoteEvent;
+
+    use Symfony\Component\RemoteEvent\Exception\ParseException;
+    use Symfony\Component\RemoteEvent\PayloadConverterInterface;
+
+    // This converter transforms a provider specific payload to
+    // our generic mailer remote event
+    class SpecificMailerProviderPayloadConverter implements PayloadConverterInterface
+    {
+        public function convert(array $payload): MailerRemoteEvent
+        {
+            // Payload contains all the information your email provider sends you
+            if (!isset($payload['mail_uid'])) {
+                throw new ParseException('This payload must contain a mail uid.');
+            }
+
+            return new MailerRemoteEvent($payload['mail_uid'], $payload, $payload['delivered'] ?? false);
+        }
+    }
+
+The ``SpecificMailerProviderPayloadConverter`` can now be injected in our request parser
+and be used to return the remote event::
+
+    namespace App\Webhook;
+
+    use App\RemoteEvent\SpecificMailerProviderPayloadConverter;
+    use Symfony\Component\HttpFoundation\ChainRequestMatcher;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\RequestMatcher\IsJsonRequestMatcher;
+    use Symfony\Component\HttpFoundation\RequestMatcher\MethodRequestMatcher;
+    use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+    use Symfony\Component\RemoteEvent\Exception\ParseException;
+    use Symfony\Component\Webhook\Client\AbstractRequestParser;
+    use Symfony\Component\Webhook\Exception\RejectWebhookException;
+
+    final class MailerWebhookParser extends AbstractRequestParser
+    {
+        public function __construct(
+            private readonly SpecificMailerProviderPayloadConverter $converter
+        ) {
+        }
+
+        protected function getRequestMatcher(): RequestMatcherInterface
+        {
+            return new ChainRequestMatcher([
+                new MethodRequestMatcher('POST'),
+                new IsJsonRequestMatcher(),
+            ]);
+        }
+
+        protected function doParse(Request $request, string $secret): ?RemoteEvent
+        {
+            $content = $request->toArray();
+
+            try {
+                // Use the new converter to create our custom remote event
+                return $this->converter->convert($content);
+            } catch (ParseException $e) {
+                throw new RejectWebhookException(406, $e->getMessage(), $e);
+            }
+        }
+    }
+
+Validate Request Preconditions
+------------------------------
+
+By using the :class:`Symfony\\Component\\HttpFoundation\\ChainRequestMatcher`,
+it is possible to build a powerful request validation chain to determine
+if the request that arrived in your webhook endpoint is valid and can
+be processed. The following matchers are available:
+
+:class:`Symfony\\Component\\HttpFoundation\\RequestMatcher\\AttributesRequestMatcher`
+    Matches request's attributes against regex
```

Comment:
I don't understand this well. Maybe we could add a short example about how to define that regex? Thanks!

Reply:
That's updated with an example. Is it better?